clayui.com/src/components/Hooks/useStickyState.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import React from 'react';
+
+const inBrowser = typeof window !== 'undefined';
+
+const localStorage = {
+	getItem(key) {
+		return inBrowser ? window.localStorage.getItem(key) : null;
+	},
+
+	setItem(key, value) {
+		if (inBrowser) {
+			window.localStorage.setItem(key, value);
+		}
+	},
+};
+
+function useStickyState(defaultValue, key) {
+	const [value, setValue] = React.useState(() => {
+		let stickyValue;
+
+		try {
+			stickyValue = localStorage.getItem(key);
+		} catch (error) {
+			stickyValue = null;
+		}
+
+		return stickyValue !== null ? JSON.parse(stickyValue) : defaultValue;
+	});
+
+	React.useEffect(() => {
+		try {
+			localStorage.setItem(key, JSON.stringify(value));
+		} catch (error) {
+			return;
+		}
```

Comment:
Personally, I'd move this one closer to (or rather, inside of) your `localStorage` implementation, so that it is closer to where the actual exception might be thrown.

(Note: didn't suggest the same for your other `try`/`catch` because `JSON.parse` can throw, so it is appropriate to have a `try`/`catch` there.)